cmd/gitops/create/dashboard/cmd.go

```diff
@@ -0,0 +1,255 @@
+		Long:  "Create a HelmRepository and HelmRelease to deploy Weave GitOps",
+		Example: `
+# Create a HelmRepository and HelmRelease to deploy Weave GitOps
+gitops create dashboard ww-gitops \
+  --password=$PASSWORD \
+  --export > ./clusters/my-cluster/weave-gitops-dashboard.yaml
+		`,
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+		PreRunE:           createDashboardCommandPreRunE(&opts.Endpoint),
+		RunE:              createDashboardCommandRunE(opts),
+		DisableAutoGenTag: true,
+	}
+
+	cmdFlags := cmd.Flags()
+
+	cmdFlags.StringVar(&flags.Username, "username", "admin", "The username of the dashboard admin user.")
+	cmdFlags.StringVar(&flags.Password, "password", "", "The password of the dashboard admin user.")
+
+	kubeConfigArgs = run.GetKubeConfigArgs()
+
+	kubeConfigArgs.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func createDashboardCommandPreRunE(endpoint *string) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		numArgs := len(args)
+
+		if numArgs == 0 {
+			return cmderrors.ErrNoName
+		}
+
+		if numArgs > 1 {
+			return cmderrors.ErrMultipleNames
+		}
+
+		name := args[0]
+		if !validateObjectName(name) {
+			return fmt.Errorf("name '%s' is invalid, it should adhere to standard defined in RFC 1123, the name can only contain alphanumeric characters or '-'", name)
+		}
+
+		return nil
+	}
+}
+
+func createDashboardCommandRunE(opts *config.Options) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		var err error
+
+		if flags.Namespace, err = cmd.Flags().GetString("namespace"); err != nil {
+			return err
+		}
+
+		kubeConfigArgs.Namespace = &flags.Namespace
+
+		if flags.KubeConfig, err = cmd.Flags().GetString("kubeconfig"); err != nil {
+			return err
+		}
+
+		if flags.Context, err = cmd.Flags().GetString("context"); err != nil {
+			return err
+		}
+
+		if flags.Export, err = cmd.Flags().GetBool("export"); err != nil {
+			return err
+		}
+
+		if flags.Timeout, err = cmd.Flags().GetDuration("timeout"); err != nil {
+			return err
+		}
+
+		var output io.Writer
+
+		if flags.Export {
+			output = &bytes.Buffer{}
+		} else {
+			output = os.Stdout
+		}
+
+		log := clilogger.NewCLILogger(output)
+
+		log.Generatef("Generating GitOps Dashboard manifests ...")
+
+		var secret string
+
+		if flags.Password != "" {
+			secret, err = run.GenerateSecret(log, flags.Password)
+			if err != nil {
+				return err
+			}
+		}
+
+		dashboardName := args[0]
+
+		adminUsername := flags.Username
+
+		if adminUsername == "" && flags.Password != "" {
+			adminUsername = defaultAdminUsername
+		}
+
+		manifests, err := run.CreateDashboardObjects(log, dashboardName, flags.Namespace, adminUsername, secret, "")
+		if err != nil {
+			return fmt.Errorf("error creating dashboard objects: %w", err)
+		} else {
+			log.Successf("Generated GitOps Dashboard manifests")
+			fmt.Println("---")
+			fmt.Println(resourceToString(manifests))
```

Comment:
Neither `flux create` or `helm install` (which is the two that comes to mind) print the manifests they're installing - if I want to know exactly what they are installing I run `flux install --export` or `helm template`.

Granted, in helm's case it's at least partially because the full output might be tens of thousands of lines, but I think that's precedent to not print the details when doing a direct install. If you want full control, use `--export`, if you want to see it work quicker just run it.

Reply:
Yes, can confirm it, checked numerous Flux commands, in about 14 commands if the export flag is set to true, the command return with printing the content. So, I will remove printing in this command in other cases, good catch, thanks!

Reply:
Added printing the manifests only when the export flag is on.